### PR TITLE
Add ability to get path stats

### PR DESCRIPTION
**Pull request by dries-c**
Motivation:
Ability to get the RTT as requested in #678

Modification:
Add ability to get the path stats using quiche_conn_path_stats

Result:
The change adds the API to get the path stats, however I wasn't sure if I should include it in QuicheConnectionStats directly.

### CODE REVIEW

File: codec-classes-quic/src/main/java/io/netty/incubator/codec/quic/QuicheQuicConnectionPathStats.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 The Netty Project
```
Comment (normanmaurer) [inline codec-classes-quic/src/main/java/io/netty/incubator/codec/quic/QuicheQuicConnectionPathStats.java:2]:
```suggestion
 * Copyright 2024 The Netty Project
```

File: codec-native-quic/src/test/java/io/netty/incubator/codec/quic/QuicConnectionPathStatsTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020 The Netty Project
```
Comment (normanmaurer) [inline codec-native-quic/src/test/java/io/netty/incubator/codec/quic/QuicConnectionPathStatsTest.java:2]:
```suggestion
 * Copyright 2024 The Netty Project
```

File: codec-classes-quic/src/main/java/io/netty/incubator/codec/quic/QuicheQuicChannel.java [hunk 3]
```diff
@@ -2050,6 +2050,33 @@ private QuicConnectionStats collectStats0(QuicheQuicConnection connection, Promi
...
+        }
+        return promise;
+    }
+
+    private void collectPathStats0(int pathIdx, Promise<QuicConnectionPathStats> promise) {
+        collectPathStats0(connection, pathIdx, promise);
+    }
+
+    private QuicConnectionPathStats collectPathStats0(QuicheQuicConnection connection, int pathIdx, Promise<QuicConnectionPathStats> promise) {
+        final Object[] stats = Quiche.quiche_conn_path_stats(connection.address(), pathIdx);
```
Comment (normanmaurer) [inline codec-classes-quic/src/main/java/io/netty/incubator/codec/quic/QuicheQuicChannel.java:2068]: check first if the connection is still valid and was not freed yet.

File: codec-native-quic/src/main/c/netty_quic_quiche.c [hunk 4]
```diff
@@ -719,6 +719,39 @@ static jobject netty_new_socket_address(JNIEnv* env, const struct sockaddr_stora
     return (*env)->NewObject(env, inetsocketaddress_class, inetsocketaddress_class_constructor, address, port);
 }
 
+static jobjectArray netty_quiche_conn_path_stats(JNIEnv* env, jclass clazz, jlong conn, jlong idx) {
+    quiche_path_stats stats = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0};
+    quiche_conn_path_stats((quiche_conn *) conn, idx, &stats);
```
Comment (normanmaurer) [inline codec-native-quic/src/main/c/netty_quic_quiche.c:724]: do we need to validate that the idx is valid first ?

Comment (dries-c) [reply codec-native-quic/src/main/c/netty_quic_quiche.c:724]: Well ideally we should, however that depends on the paths_count of quiche_stats and there doesn't seem like there's another way to validate it.

Comment (normanmaurer) [inline codec-native-quic/src/main/c/netty_quic_quiche.c:724]:
```suggestion
    if (quiche_conn_path_stats((quiche_conn *) conn, idx, &stats) != 0) {
        // The idx is not valid. 
        return NULL;
    }
```

**Review by normanmaurer — CHANGES_REQUESTED** [review]
Almost there

**Comment by normanmaurer** [comment]
@dries-c thanks!

**Pull request merged**